### ISSUE TITLE
Set item filter through a GUI instead of item frames

### ISSUE THREAD

**Issue by ReySniperio** [flokol120/Spigot-Item-Chest-Sorter#49]
Item frames can become laggy on a server, it would be better if you can set a filter through a GUI.

**Comment by flokol120** [comment]
@ReySniperio Please elaborate, what you mean exactly by _filter through a GUI_. In which situations are item frames "laggy" for you? Spigot/Paper already has a few optimizations in place to limit performance impact of item frames such as a lower tick rate. I've never experienced any problems with items frames myself. That's why I am curios how you managed to have a performance issue with them :D

**Comment by ReySniperio** [comment]
Lots of double chest and item frames always have been laggy. Maybe not if you only use them for the sorter but we use them also for decorating etc. Never the less a option for setting filters on a chest through a GUI would be more convenient. For example clicking a chest with a "filter hoe" will open a gui where you can set your filter by dragging in the specific items you want.

**Comment by flokol120** [comment]
Alright, but chests cannot be eliminated out of the equation obviously :D But lets not focus on performance too much, I understand you feature request UI-wise. Unfortunately, I am not really able to fathom how you imagine this GUI would look like. Could you please be more specific. Maybe you could use a Screenshot of a similar use case or even mocking it up.

**Comment by ReySniperio** [comment]
![2023-07-25_16 21 08](https://github.com/flokol120/Spigot-Item-Chest-Sorter/assets/77800671/b6e6bcdf-166a-4d5b-afa3-62b16ba2df5d)

If you click on a chest with a "_filter hoe_" a GUI appear where you can set your flters for a specific chest.
This would be better in my eyes. Now you have to configure sets or you have to put loads of itemframes on a chest.
In case of sets, not every player wants it like the sets are configured.

PS: dont mind the icons in my inventory thats a mod.

**Comment by flokol120** [comment]
Ahh, thanks for the clarification. Now I get it :). I hate to brek it, though this would won't be that easy to implent, as this would mean I had to change some core elements of the plugin. But in general I really like this idea and I'll come back to it as soon as I have more free time. 

TL; DR: I cannot give an ETA but I'll tend to it sooner or later.

In the end I would however combine the simplicity of the item frame approach and the GUI approach by displaying the first item of the GUI within the item frame. This will ensure backwards compatibility.

**Comment by AE720** [comment]
> Lots of double chest and item frames always have been laggy. Maybe not if you only use them for the sorter but we use them also for decorating etc. Never the less a option for setting filters on a chest through a GUI would be more convenient. For example clicking a chest with a "filter hoe" will open a gui where you can set your filter by dragging in the specific items you want.

A lot of that is the nature in how Minecraft handles timings. 

If you set  hanging-tick-frequency to 2147483647 in spigot.yml, that will help and for chests, they produce added ticks for things like animations. Mods (not plugins) can disable that but you can also just use barrels and that'll fix that issue.
